ui/src/data_explorer/components/VisHeader.js

```diff
@@ -2,34 +2,55 @@ import React, {PropTypes} from 'react'
 import classnames from 'classnames'
 import _ from 'lodash'
 
-const VisHeader = ({views, view, onToggleView, name}) =>
+import {fetchTimeSeriesAsync} from 'shared/actions/timeSeries'
+import resultsToCSV from 'src/shared/parsing/resultsToCSV.js'
+import download from 'src/external/download.js'
+
+const getCSV = query => async () => {
+  try {
+    const {results} = await fetchTimeSeriesAsync({source: query.host, query})
+    const {name, CSVString} = resultsToCSV(results)
+    download(CSVString, `${name}.csv`, 'text/plain')
+  } catch (error) {
+    console.error(error)
+  }
```

Comment:
Let's also publish and error to the user here to let them know something went wrong